slave/process-kerberos-renewal-principals/bin/process-kerberos_renewal_principals.sh

```diff
@@ -1,25 +1,45 @@
 #!/bin/bash
-
 PROTOCOL_VERSION='3.0.0'
 
 function process {
 
 	DST_DIR="/etc/heimdal-kdc/krb525d.d/"
-	DST_FILE="${SERVICE}.conf"
-	FROM_PERUN="${WORK_DIR}/${SERVICE}"
+	FROM_PERUN_DIR="${WORK_DIR}/kerberos_renewal_principals/"
 
 	### Status codes
-	I_CHANGED=(0 "${DST_FILE} updated")
-	E_NOT_CHANGE=(50 "Cannot copy file ${FROM_PERUN} to ${DST_FILE}")
+	I_EVERYTHING_OK=(0 'All files has been updated.')
+	E_FINISHED_WITH_ERRORS=(50 'Slave script finished with errors!')
+
+	ERROR=0
 
 	create_lock
 
-	cp "${FROM_PERUN}" "${DST_DIR}/${DST_FILE}"
+	# Delete all files from destination directory
+	if [[ $(find $DST_DIR -mindepth 1 -maxdepth 1) ]]; then
+		if ! rm "$DST_DIR"*; then
```

Comment:
I'm afraid, that destination folder contains also other files of service configuration, so we should only delete files with `kerberos_renewal_principals` prefix in the destination folder.